### PR TITLE
fix: fix data types for lane ids

### PR DESCRIPTION
**Pull request by mitsudome-r**
## Description
Lanelet2 uses int64 as data type to store ids.
However, there are inconsistency in control/planning modules to store the data.

I have searched the code in Autoware with keyword "lane_id" and "lanelet_id" and tried to fix as much as possible to use int64_t or lanelet::Id for data type declaration.

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by kosuke55 — APPROVED** [review]

**Review by soblin — APPROVED** [review]

**Review by takayuki5168 — APPROVED** [review]

**Review by satoshi-ota — APPROVED** [review]

**Comment by mitsudome-r** [comment]
Thanks for people who approved the PR.

I still need the review from the following package maintainers:
- lane_departure_checker @kyoichi-sugahara 
- behavior_velocity_planner_common @tkimura4, @shmpwk, @isamu-takagi 
- planning_test_utils @kyoichi-sugahara , @TakaHoribe

**Review by shmpwk — APPROVED** [review]

**Review by isamu-takagi — APPROVED** [review]

**Review by tkimura4 — APPROVED** [review]

**Review by taikitanaka3 — APPROVED** [review]

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM!
So sorry for late approve :bow:

**Pull request merged**
